_ap/17/index.adoc

```diff
@@ -0,0 +1,59 @@
+---
+num: 17
+tags:
+  - "tls"
+  - "certificates"
+  - "dns"
+  - "Letsencrypt"
+title: "Automatic management of certificates per managed instance using Letsencrypt"
+status: "Accepted"
+authors:
+  - "Manyanda Chitimbo"
+---
+
+## Intent
+
+Define a pattern for the automatic management of certificates per managed instance using Letsencrypt
+
+## Motivation
+
+When a managed service is deployed on the data plane cluster its routes have be reachable via TLS.
+It means that a managed service deployed in the data plane has to provide a way to the clients (end users) to verify and validate the TLS connection.
+
+## Applicability
+
+This pattern applies to any service deployed on a dataplane that wants a managed service routes to be reachable via TLS
+
+## Description
+
+A managed service will use https://Letsencrypt.org[Letsencrypt] to managed certificates on the fly i.e automatic and thus removing the need for manual intervention and the toil related to this.
+
+To avoid sharing of certificates between deployed instances, managed service implementors are recommended to generate a dedicated certificate per managed instance.
+
+As an example, if the root domain is `myservice.bf2.dev`, then a dedicated certificate will be generated for each new managed instance under a subdomain `<managed-instance-id>.myservice.bf2.dev`. The `<managed-instance-id>` is the unique id of the managed instance.
+
+Furthermore, all routes for the managed services deployed on the the data plane will be of the form `<managed-instance-id>.myservice.bf2.dev`.
+If we take the Managed Kafka Service as an example, it generates a dedicated wildcard certificate per managed Kafka instance - see link:../../adr/90[ADR-90].
+All the Kafka routes are of the form `<unique-route-slug>.<kafka-instance-id>.kafka.bf2.dev` e.g `boostrapserverhost.123456.kafka.bf2.dev`.
```

Comment:
It feels like this section should explicitly call out that the subdomain used for an instance must be independent of the dataplane in which it (currently) resides.

Reply:
Thanks. I'll add that in.